pkg/odo/util/completion/completion.go

```diff
@@ -76,3 +80,50 @@ func GetCommandFlagHandler(command *cobra.Command, flag string) (predictor compl
 	predictor, ok = completionHandlers[getCommandFlagCompletionHandlerKey(command, flag)]
 	return
 }
+
+func getCommandsAndFlags(args []string, c *cobra.Command) ([]string, map[string]string) {
```

Comment:
I believe this method can be implemented in a much simpler way:
```go
func getCommandsAndFlags(args []string, c *cobra.Command) ([]string, map[string]string) {
	if len(args) == 0 {
		return args, nil
	}
	c.ParseFlags(args)
	flags := c.Flags()

	cmds := flags.Args()
	setFlags := make(map[string]string)
	flags.Visit(func(i *flag.Flag) {
		if i.Value.Type() != "bool" {
			setFlags[i.Name]=i.Value.String()
		}
	})

	return cmds, setFlags
}
```

Reply:
As a result, `isBooleanFlag` can also be removed.

Reply:
Wow! I didn't notice there was a flags.Args() function for the job. Thanks :+1:  Will try it out